### PR TITLE
update cli to check for commit hash diff

### PR DESCRIPTION
**Pull request by jhoetter**
To test, run `pip install -e .` inside the repository to build the newest CLI. Afterward, you can run `refinery start`

This should let you know once running `refinery start` that there is an updated version available, which you can pull via user input `y`.

### CODE REVIEW

**Comment by jhoetter** [comment]
implements #29

**Comment by JWittmeyer** [comment]
cli version should be 1.1.0 or 1.0.3

- [ ] resolved

File: refinery/cli.py [hunk 1]
```diff
@@ -16,7 +17,32 @@ def start(cur_dir: str):
...
+
+            if sha_local != sha_remote:
+                msg.info(
+                    "A new version of refinery is available. Should this be pulled? (y/n)"
+                )
+                user_input = input("> ")
+                if user_input == "y":
+                    repo_origin = repo.remotes.origin
+                    repo_origin.pull()
+                    msg.good(f"refinery has been updated to commit {sha_remote}.")
```
Comment (JWittmeyer) [inline refinery/cli.py:42]: not sure if gitpython work with it but a pull can run into some issues e.g. if you have a fork and uncommitted changes -> maybe we can check it?

**Comment by JWittmeyer** [comment]
Currently only the hash is checked. This might result in some unforeseen issues e.g. for a fork.
One way to work around it could be to only do the check if the use is on main branch. 
Another option could be to check commits ahead/behind with:
`git rev-list --left-right --count main...hash-check-cli`

however this is a cli command (subprocess) not a gitpyhon one.

- [ ] resolve
- [ ] discussed
- [ ] ignored

File: refinery/cli.py (hunk 1 above)
Comment (jhoetter) [reply refinery/cli.py]: I have found a solution and implemented it :)

**Review by JWittmeyer — APPROVED** [review]

**Pull request merged**